### PR TITLE
Add MLA support to vLLM patch

### PR DESCRIPTION
**Pull request by rohansjoshi**
### What does this PR do?

Builds on [vLLM integration PR ](https://github.com/NVIDIA/Model-Optimizer/pull/1127/changes):
- Adds support for threshold scale factors (instead of fixed thresholds) in skip softmax Triton kernel
- Adds MLA support in vLLM patch

Type of change: new feature

### Usage

```
SPARSE_ATTN_CFG=SPARSE_SOFTMAX_DEFAULT python
examples/vllm_serve/vllm_serve_sparse_attn.py \
    nvidia/DeepSeek-R1-NVFP4
```

## Summary by CodeRabbit

* **New Features**
  * Added vLLM server integration with sparse attention support, including example scripts and worker implementations for efficient inference.
  * Introduced paged KV-cache support for Triton flash-attention kernels.
  * Added skip-softmax scale-factor threshold mode for fine-grained sparsity control.

* **Dependencies**
  * Added `uvloop` and `vllm==0.20.1` dependencies.

* **Tests**
  * Added comprehensive test coverage for paged KV-cache functionality and skip-softmax scale-factor behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/NVIDIA/Model-Optimizer/pull/1499)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -49,6 +49,8 @@ dependencies = [
     "rich",
     "safetensors",
     "scipy",
+    "uvloop>=0.22.1",
+    "vllm==0.20.1",
```
[the remark this answers is not in the document]
Comment (shengliangxu) [reply pyproject.toml:53]: oh, do not add them here, add a requirements.txt into vllm_serve example

File: modelopt/torch/sparsity/attention_sparsity/plugins/__init__.py [hunk 2]
```diff
@@ -15,6 +15,8 @@
 
 """Plugins for sparse attention integration with various frameworks."""
 
+from modelopt.torch.utils import import_plugin
```
Comment (shengliangxu) [inline modelopt/torch/sparsity/attention_sparsity/plugins/__init__.py:18]: seems dead code?
